File: 3979-maximum-valid-pair-sum/3979-maximum-valid-pair-sum.java
public class Solution {
    public int maxValidPairSum(int[] nums, int k) {
        int maxSum = Integer.MIN_VALUE;
        int maxI = Integer.MIN_VALUE;

        for (int j = k; j < nums.length; j++) {
            maxI = Math.max(maxI, nums[j - k]);
            maxSum = Math.max(maxSum, maxI + nums[j]);
        }

        return maxSum;
    }
}

File: 3979-maximum-valid-pair-sum/3979-maximum-valid-pair-sum.py
class Solution(object):
    def maxValidPairSum(self, nums, k):
        max_pair_sum = -float('inf')
        max_i_val = -float('inf')

        for j in range(k, len(nums)):
            max_i_val = max(max_i_val, nums[j - k])
            max_pair_sum = max(max_pair_sum, max_i_val + nums[j])

        return max_pair_sum
